packages/react-native-web/src/modules/ImageLoader/index.js

```diff
@@ -111,37 +119,78 @@ const ImageLoader = {
       if (typeof failure === 'function') {
         failure();
       }
-      ImageLoader.abort(requestId);
+      ImageLoader.release(requestId);
       clearInterval(interval);
     }
   },
-  has(uri: string): boolean {
+  has(uri: ?string): boolean {
     return ImageUriCache.has(uri);
   },
-  load(uri: string, onLoad: Function, onError: Function): number {
+  load(
+    source: ImageSource,
+    onLoad: (ImageResult) => void,
+    onError: Function
+  ): number {
     id += 1;
     const image = new window.Image();
-    image.onerror = onError;
-    image.onload = (e) => {
+
+    const handleLoad = () => {
       // avoid blocking the main thread
-      const onDecode = () => onLoad({ nativeEvent: e });
-      if (typeof image.decode === 'function') {
-        // Safari currently throws exceptions when decoding svgs.
-        // We want to catch that error and allow the load handler
-        // to be forwarded to the onLoad handler in this case
-        image.decode().then(onDecode, onDecode);
-      } else {
-        setTimeout(onDecode, 0);
-      }
+      const onDecode = () => {
+        ImageUriCache.add(source.uri);
+        onLoad({
+          source: {
+            uri: image.src,
+            width: image.naturalWidth,
+            height: image.naturalHeight
+          }
+        });
```

Comment:
This is a BREAKING CHANGE 
The `nativeEvent` from the `onLoad` callback is now `{ nativeEvent: { source: { width, height, uri } } }`
I can preserve the old structure where `nativeEvent` is just the `Event` object from `image.onload`, but I'll have to add these width/height/uri properties under a different key and they won't match the same interface as used in react-native: https://reactnative.dev/docs/image#onload

Do you want to change the code to avoid a breaking change or do you want it to match `react-native` interface?